bindings/go/scip/symbol.go

```diff
@@ -8,12 +8,41 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
+// IsGlobalSymbol returns true if the symbol is obviously not a local symbol.
+//
+// CAUTION: Does not perform full validation of the symbol string's contents.
 func IsGlobalSymbol(symbol string) bool {
```

Comment:
I think it's inconsistent that `IsGlobalSymbol` does no validation while `IsLocalSymbol` does validation. Either both of them validate or neither of them do. I propose neither of them do validation, and we add separate `IsWellFormedLocalSymbol`/`IsWellFormedGlobalSymbol` functions validate.

Validation can become a performance bottleneck if you run the same validation on every single symbol in a critical path. Applications should be able to have if conditions to switch behavior for global/local symbols without performing validation (because the payload has been validated elsewhere in the pipeline)

Reply:
Removed the validation from `IsLocalSymbol`